cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1246,6 +1246,39 @@ const char* convert_fromUTF8_binary(gdv_int64 context, const char* bin_in, gdv_i
   return ret;
 }
 
+FORCE_INLINE
+const char* convert_replace_invalid_fromUTF8_binary(
+    gdv_int64 context, const char* text_in, gdv_int32 text_len,
+    const char* char_to_replace, gdv_int32 /*char_to_replace_len*/, gdv_int32* out_len) {
+  // actually the convert_replace function replaces the invalid bytes with a single byte
+  // so the output length will be the same as the input length
```

Comment:
If you are assuming it is single byte char, add a check to validate that the char_length of char_to_replace is 1. If its greater than one, return error saying not supported

Reply:
Added!